core/controllers/story_viewer.py

```diff
@@ -183,6 +183,10 @@ def _record_node_completion(
                 completed_node.id for completed_node in completed_nodes]
 
             for node in ordered_nodes:
+                if node.id == node_id and node.exploration_id is not None:
```

Comment:
I'm not sure this is the right place to do this. The place where the action is happening isn't actually in the story viewer.

Probably the way to go about it is to see where the "exploration finished" handler is, and update its logic to do one thing in the story context and a different thing in the non-story context.

Please also add tests for this new functionality -- thanks.